### PR TITLE
Add DNS Records subpage and add support to custom headers

### LINKED ISSUE #97279: Show the "Manage DNS Records" page
Create page that shows the DNS records.

Image

- [ ] The page looks as new design
- [ ] The `Import BIND file` button works as expected
- [ ] The `+ Add a record` button works as expected

### PR DESCRIPTION
**Pull request by phcp**
Closes https://github.com/Automattic/wp-calypso/issues/97279

**Should be rebased and merged after completing https://github.com/Automattic/wp-calypso/pull/97696.**

## Proposed Changes

* Added DNS Records subpage.
* Added `CustomHeader` option to the `SubpageWrapper` component.
* Created a custom header for the Add Forwarding Email page (including breadcrumb).
* Update header styles.
* Changes didn't affect visually the Edit Contact Info page.

Screenshot 2024-12-20 at 09 12 21

## Why are these changes being made?

* This is part of the Domain Management Revamp project: pfuQfP-13x-p2

## Testing Instructions

* Apply this PR to your local
* Go to http://calypso.localhost:3000/domains/manage?flags=calypso/all-domain-management

**Add Forwarding Email page**
* Click on a domain that doesn't have emails configured
* Click on the Email tab
* Scroll to the bottom of the page and click the Email Forwarding link
* You should be redirected to the Add Forwarding Email page
* Check if the page is rendered as expected
* Make sure the header is also rendered correctly. Test on different screen sizes.
* Perform regression tests on the old page. It should continue working as before without any visual changes

**DNS Records page**
* On the Domains overview, scroll down and expand the DNS Records card.
* Click on Manage
* You should be redirected to the revamped DNS Records page
* Check if the page is rendered as expected. Also, check on multiple screen sizes.
* Perform regression tests on the page, testing all the actions.
* Go to the old version of the page and make sure it's working as expected as well.

**Edit Contact Info page**
* Perform regression tests on the page. You can follow instructions from https://github.com/Automattic/wp-calypso/pull/97562.
* The Edit Contact Info page should look as intended on Miro's PR. No visual change was applied there.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/domains/domain-management/subpage-wrapper/style.scss [hunk 1]
```diff
@@ -6,6 +7,11 @@
 	flex-direction: column;
 	height: 100%;
 
+    .button:not(.is-borderless) {
```
Comment (bogiii) [inline client/my-sites/domains/domain-management/subpage-wrapper/style.scss]: This block overrides other button styles, see the example:

Screenshot 2024-12-25 at 14 49 41

I would rather keep the default button style than override it.

**Review by bogiii — COMMENTED** [review]
Comparing with the figma design, there is a DNS records list item menu alignment issue.
There is a `right: 8px` rule. It should be removed.

Screenshot 2024-12-25 at 15 13 47

**Review by bogiii — APPROVED** [review]
I fixed a few styling issues and tested the changes that were introduced.
Overall, everything looks good.  🚢 

The suggestion for the future: we should separate large PRs into smaller chunks of changes for easier review.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17079840

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @phcp for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17079840).

**Pull request merged**